Upgrade JetStack Cert-Manager to v0.15.2

bmelbourne opened the pull request:
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

This PR upgrades the deployment of JetStack Cert-Manager to v0.15.2.

**Which issue(s) this PR fixes**:
Fixes #5946 

**Special notes for your reviewer**:
For additional information, refer to `roles\kubernetes-apps\ingress_controller\cert_manager\README.md`.

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
Action required: Refer to roles\kubernetes-apps\ingress_controller\cert_manager\README.md prior to upgrading JetStack Cert-Manager in your exisitng Kubernetes cluster.
```

bmelbourne commented:
/assign @mattymo

Miouge1 commented:
CI fails with:

```
fatal: [instance-1]: FAILED! => {"msg": "The conditional check 'item in cert_manager_webhook_pods_created.stdout' failed. The error was: error while evaluating conditional (item in cert_manager_webhook_pods_created.stdout): Unable to look up a name or access an attribute in template string ({% if item in cert_manager_webhook_pods_created.stdout %} True {% else %} False {% endif %}).\nMake sure your variable name does not contain invalid characters like '-': argument of type 'AnsibleUndefined' is not iterable"}
```

Is that something you can look into?

bmelbourne commented:
> CI fails with:
> 
> ```
> fatal: [instance-1]: FAILED! => {"msg": "The conditional check 'item in cert_manager_webhook_pods_created.stdout' failed. The error was: error while evaluating conditional (item in cert_manager_webhook_pods_created.stdout): Unable to look up a name or access an attribute in template string ({% if item in cert_manager_webhook_pods_created.stdout %} True {% else %} False {% endif %}).\nMake sure your variable name does not contain invalid characters like '-': argument of type 'AnsibleUndefined' is not iterable"}
> ```
> 
> Is that something you can look into?

The error appears to be isolated to VMs running CentOS 7 only.

I'm been able to reproduce the error by changing `$os ||= "centos"` in the default Vagrantfile.

Hopefully, I'll be able to find a fix for the issue over the next few days.

bmelbourne commented:
/retest

bmelbourne commented:
> > CI fails with:
> > ```
> > fatal: [instance-1]: FAILED! => {"msg": "The conditional check 'item in cert_manager_webhook_pods_created.stdout' failed. The error was: error while evaluating conditional (item in cert_manager_webhook_pods_created.stdout): Unable to look up a name or access an attribute in template string ({% if item in cert_manager_webhook_pods_created.stdout %} True {% else %} False {% endif %}).\nMake sure your variable name does not contain invalid characters like '-': argument of type 'AnsibleUndefined' is not iterable"}
> > ```
> > 
> > 
> > Is that something you can look into?
> 
> The error appears to be isolated to VMs running CentOS 7 only.
> 
> I'm been able to reproduce the error by changing `$os ||= "centos"` in the default Vagrantfile.
> 
> Hopefully, I'll be able to find a fix for the issue over the next few days.

@Miouge1 
I've refactored and tested the Ansible code to get it to work on CentOS 7.x and Python 2.7.5.
Can you retrigger the CI tests?

floryut commented:
> > > CI fails with:
> > > ```
> > > fatal: [instance-1]: FAILED! => {"msg": "The conditional check 'item in cert_manager_webhook_pods_created.stdout' failed. The error was: error while evaluating conditional (item in cert_manager_webhook_pods_created.stdout): Unable to look up a name or access an attribute in template string ({% if item in cert_manager_webhook_pods_created.stdout %} True {% else %} False {% endif %}).\nMake sure your variable name does not contain invalid characters like '-': argument of type 'AnsibleUndefined' is not iterable"}
> > > ```
> > > 
> > > 
> > > Is that something you can look into?
> > 
> > 
> > The error appears to be isolated to VMs running CentOS 7 only.
> > I'm been able to reproduce the error by changing `$os ||= "centos"` in the default Vagrantfile.
> > Hopefully, I'll be able to find a fix for the issue over the next few days.
> 
> @Miouge1
> I've refactored and tested the Ansible code to get it to work on CentOS 7.x and Python 2.7.5.
> Can you retrigger the CI tests?

CI is reporting a merge issue on your PR, could you check ?

bmelbourne commented:
> > > > CI fails with:
> > > > ```
> > > > fatal: [instance-1]: FAILED! => {"msg": "The conditional check 'item in cert_manager_webhook_pods_created.stdout' failed. The error was: error while evaluating conditional (item in cert_manager_webhook_pods_created.stdout): Unable to look up a name or access an attribute in template string ({% if item in cert_manager_webhook_pods_created.stdout %} True {% else %} False {% endif %}).\nMake sure your variable name does not contain invalid characters like '-': argument of type 'AnsibleUndefined' is not iterable"}
> > > > ```
> > > > 
> > > > 
> > > > Is that something you can look into?
> > > 
> > > 
> > > The error appears to be isolated to VMs running CentOS 7 only.
> > > I'm been able to reproduce the error by changing `$os ||= "centos"` in the default Vagrantfile.
> > > Hopefully, I'll be able to find a fix for the issue over the next few days.
> > 
> > 
> > @Miouge1
> > I've refactored and tested the Ansible code to get it to work on CentOS 7.x and Python 2.7.5.
> > Can you retrigger the CI tests?
> 
> CI is reporting a merge issue on your PR, could you check ?

I've already fixed the errors that are being reported by the CI tests but for some reason, they're still running an old commit.

Should I delete this PR and resubmit? It seems have got itself in a bit of a mess.

bmelbourne commented:
> > > > > > CI fails with:
> > > > > > ```
> > > > > > fatal: [instance-1]: FAILED! => {"msg": "The conditional check 'item in cert_manager_webhook_pods_created.stdout' failed. The error was: error while evaluating conditional (item in cert_manager_webhook_pods_created.stdout): Unable to look up a name or access an attribute in template string ({% if item in cert_manager_webhook_pods_created.stdout %} True {% else %} False {% endif %}).\nMake sure your variable name does not contain invalid characters like '-': argument of type 'AnsibleUndefined' is not iterable"}
> > > > > > ```
> > > > > > 
> > > > > > 
> > > > > > Is that something you can look into?
> > > > > 
> > > > > 
> > > > > The error appears to be isolated to VMs running CentOS 7 only.
> > > > > I'm been able to reproduce the error by changing `$os ||= "centos"` in the default Vagrantfile.
> > > > > Hopefully, I'll be able to find a fix for the issue over the next few days.
> > > > 
> > > > 
> > > > @Miouge1
> > > > I've refactored and tested the Ansible code to get it to work on CentOS 7.x and Python 2.7.5.
> > > > Can you retrigger the CI tests?
> > > 
> > > 
> > > CI is reporting a merge issue on your PR, could you check ?
> > 
> > 
> > @floryut
> > I've already fixed the errors that are being reported by the CI tests but for some reason, they're still running an old commit.
> > Should I delete this PR and resubmit? It seems have got itself in a bit of a mess.
> 
> No :) can you try to rebase master locally and see what happen ?

@floryut 
Still no luck, I've rebased with master locally and resolved conflicts 3 times now, and created another commit.
If it's not possible to kick-off clean CI tests, can you delete the PR and I'll raise a new one with a single commit.
Hopefully, the CI tests will run again and I can see any new errors.

floryut commented:
/close

floryut commented:
/reopen

floryut commented:
> > > > > > > CI fails with:
> > > > > > > ```
> > > > > > > fatal: [instance-1]: FAILED! => {"msg": "The conditional check 'item in cert_manager_webhook_pods_created.stdout' failed. The error was: error while evaluating conditional (item in cert_manager_webhook_pods_created.stdout): Unable to look up a name or access an attribute in template string ({% if item in cert_manager_webhook_pods_created.stdout %} True {% else %} False {% endif %}).\nMake sure your variable name does not contain invalid characters like '-': argument of type 'AnsibleUndefined' is not iterable"}
> > > > > > > ```
> > > > > > > 
> > > > > > > 
> > > > > > > Is that something you can look into?
> > > > > > 
> > > > > > 
> > > > > > The error appears to be isolated to VMs running CentOS 7 only.
> > > > > > I'm been able to reproduce the error by changing `$os ||= "centos"` in the default Vagrantfile.
> > > > > > Hopefully, I'll be able to find a fix for the issue over the next few days.
> > > > > 
> > > > > 
> > > > > @Miouge1
> > > > > I've refactored and tested the Ansible code to get it to work on CentOS 7.x and Python 2.7.5.
> > > > > Can you retrigger the CI tests?
> > > > 
> > > > 
> > > > CI is reporting a merge issue on your PR, could you check ?
> > > 
> > > 
> > > @floryut
> > > I've already fixed the errors that are being reported by the CI tests but for some reason, they're still running an old commit.
> > > Should I delete this PR and resubmit? It seems have got itself in a bit of a mess.
> > 
> > 
> > No :) can you try to rebase master locally and see what happen ?
> 
> @floryut
> Still no luck, I've rebased with master locally and resolved conflicts 3 times now, and created another commit.
> If it's not possible to kick-off clean CI tests, can you delete the PR and I'll raise a new one with a single commit.
> Hopefully, the CI tests will run again and I can see any new errors.

Well I've checkout your branch and tried a rebase with spray/master and got some conflict, exactly what the CI got :)

floryut commented:
> > > > > > > CI fails with:
> > > > > > > ```
> > > > > > > fatal: [instance-1]: FAILED! => {"msg": "The conditional check 'item in cert_manager_webhook_pods_created.stdout' failed. The error was: error while evaluating conditional (item in cert_manager_webhook_pods_created.stdout): Unable to look up a name or access an attribute in template string ({% if item in cert_manager_webhook_pods_created.stdout %} True {% else %} False {% endif %}).\nMake sure your variable name does not contain invalid characters like '-': argument of type 'AnsibleUndefined' is not iterable"}
> > > > > > > ```
> > > > > > > 
> > > > > > > 
> > > > > > > Is that something you can look into?
> > > > > > 
> > > > > > 
> > > > > > The error appears to be isolated to VMs running CentOS 7 only.
> > > > > > I'm been able to reproduce the error by changing `$os ||= "centos"` in the default Vagrantfile.
> > > > > > Hopefully, I'll be able to find a fix for the issue over the next few days.
> > > > > 
> > > > > 
> > > > > @Miouge1
> > > > > I've refactored and tested the Ansible code to get it to work on CentOS 7.x and Python 2.7.5.
> > > > > Can you retrigger the CI tests?
> > > > 
> > > > 
> > > > CI is reporting a merge issue on your PR, could you check ?
> > > 
> > > 
> > > @floryut
> > > I've already fixed the errors that are being reported by the CI tests but for some reason, they're still running an old commit.
> > > Should I delete this PR and resubmit? It seems have got itself in a bit of a mess.
> > 
> > 
> > No :) can you try to rebase master locally and see what happen ?
> 
> @floryut
> Still no luck, I've rebased with master locally and resolved conflicts 3 times now, and created another commit.
> If it's not possible to kick-off clean CI tests, can you delete the PR and I'll raise a new one with a single commit.
> Hopefully, the CI tests will run again and I can see any new errors.

Well we can create another PR but this won't change anything.
You can delete your branch locally, clean it up and push force it.

floryut commented:
Closing in favor of #6414 ... :)
/close

bmelbourne commented:
@floryut 
Thanks for closing this PR.
Somewhere during the updates between my fork branch and the upstream PR branch, the commits have become corrupted.
I've been having some intermittent home network issues over the past week, so this might have contributed to the issue.
Going forward, I'll wait for the all the tests to complete before applying any further commits.
